beets/ui/__init__.py

```diff
@@ -367,6 +367,31 @@ def input_yn(prompt, require=False):
     return sel == 'y'
 
 
+def input_select_items(prompt, items, rep):
+    """Prompts the user to use all, none or some of the items
+    Will return the list of items the user selected
+    prompt: prompt to use for all and for selective choice
+    items: full list of items
+    rep: function which represents an item to the user
+         is called with the item as argument
+         function is responsive for newline at input
+    """
+    out_items = []
+    choice = input_options(
+        ('y', 'n', 's'), False,
+        '%s (Yes/No/Selective)?' % prompt)
+    print()
+    if choice == 'y':
+        out_items = items
+    elif choice == 's':
+        for item in items:
+            rep(item)
+            if input_yn('%s (y/n)?' % prompt, True):
```

Comment:
yeah, thats right. I would suggest to press CTRL-C in this case and try it again.

The best will be to implement the interface in a way `git checkout -p` does.
```
Apply this hunk to index and worktree [y,n,q,a,d,/,e,?]? ?
y - apply this hunk to index and worktree
n - do not apply this hunk to index and worktree
q - quit; do not apply this hunk nor any of the remaining ones
a - apply this hunk and all later hunks in the file
d - do not apply this hunk nor any of the later hunks in the file
g - select a hunk to go to
/ - search for a hunk matching the given regex
j - leave this hunk undecided, see next undecided hunk
J - leave this hunk undecided, see next hunk
k - leave this hunk undecided, see previous undecided hunk
K - leave this hunk undecided, see previous hunk
s - split the current hunk into smaller hunks
e - manually edit the current hunk
? - print help
```

I would suggest these commands:
- y - yes
- n - no
- a - print all following and ask again
- q - quit and revert selected ones
- ? - print help

please leave comment whether this attracts to you